src/jit/codegencommon.cpp

```diff
@@ -4282,7 +4282,14 @@ void CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg, bool* pXtraRegClobbere
 #endif // defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
         {
             // Bingo - add it to our table
-            regArgNum = genMapRegNumToRegArgNum(varDsc->lvArgReg, regType);
+            if (!compiler->opts.compUseSoftFP)
+            {
+                regArgNum = genMapRegNumToRegArgNum(varDsc->lvArgReg, regType);
+            }
+            else
+            {
+                regArgNum = genMapIntRegNumToRegArgNum(varDsc->lvArgReg);
+            }
 
```

Comment:
`regState->rsCalleeRegArgMaskLiveIn` isn't empty for floating-point register in `compUseSoftFP` mode. And  compiler crashes inside `genMapRegNumToRegArgNum` call.
So if it should be empty bug in another place.